dask_glm/utils.py

```diff
@@ -137,18 +138,32 @@ def sum(A):
     return A.sum()
 
 
+def is_dask_array_sparse(X):
+    """
+    Check using _meta if a dask array contains sparse arrays
+    """
+    return isinstance(X._meta, sparse.COO) or isinstance(X._meta, sparse.DOK)
+
+
 @dispatch(np.ndarray)
 def add_intercept(X):
     return np.concatenate([X, np.ones((X.shape[0], 1))], axis=1)
 
 
+@dispatch((sparse.COO, sparse.DOK))
+def add_intercept(X):
+    return sparse.concatenate([X, sparse.COO(np.ones((X.shape[0], 1)))], axis=1)
```

Comment:
Is there any advantage to making the `ones` we create the same sparse format as `X`? Right now, we may concat a COO to a DOK, which IIUC results in a COO.

Reply:
`concatenate(DOK, DOK)` is also `COO` so I think this is fine.

Reply:
`concatenate([all_DOK])` is also `COO`, so I'd recommend using `COO`.